packages/alchemy/src/react/hooks/useSmartAccountClient.ts

```diff
@@ -0,0 +1,83 @@
+"use client";
+
+import {
+  accountLoupeActions,
+  lightAccountClientActions,
+  multiOwnerPluginActions,
+  pluginManagerActions,
+  type AccountLoupeActions,
+  type LightAccount,
+  type LightAccountClientActions,
+  type MultiOwnerModularAccount,
+  type MultiOwnerPluginActions,
+  type PluginManagerActions,
+} from "@alchemy/aa-accounts";
+import type { Chain, CustomTransport, Transport } from "viem";
+import { createAlchemySmartAccountClientFromRpcClient } from "../../client/internal/smartAccountClientFromRpc.js";
+import type {
+  AlchemySmartAccountClient,
+  AlchemySmartAccountClientConfig,
+  BaseAlchemyActions,
+} from "../../client/smartAccountClient";
+import type { SupportedAccount, SupportedAccountTypes } from "../../config";
+import type { AlchemySigner } from "../../signer";
+import { useBundlerClient } from "./useBundlerClient.js";
+
+export type UseSmartAccountClientProps<TAccount extends SupportedAccountTypes> =
+  Omit<
+    AlchemySmartAccountClientConfig<
+      Transport,
+      Chain,
+      SupportedAccount<TAccount>
+    >,
+    "rpcUrl" | "chain" | "apiKey" | "jwt" | "account"
+  > & { account: SupportedAccount<TAccount> };
+
+export type ClientActions<TAccount extends SupportedAccountTypes> =
+  TAccount extends "LightAccount"
+    ? BaseAlchemyActions<Chain, LightAccount<AlchemySigner>> &
```

Comment:
this isn't necessary in @adam-alchemy's multisig branch because we simplify it a bit to not require passing in the `BaseActions`

Reply:
so will need to update here once that's merged in (or fix in his branch if this is merged in first)